hybridq/noise/channel/utils.py

```diff
@@ -178,5 +185,11 @@ def choi_matrix(channel: MatrixChannel,
 
 
 def _channel_dim(channel):
-    # dimension (assume all Kraus' have same shape)
-    return channel.Kraus.gates[0][0].matrix().shape[0]
+    # map() gives the dimension squared of the channel
+    full_dims = channel.map().shape
```

Comment:
I modified this using `map()` to extract the dimension (which is required for the Choi matrix). Before it was too restrictive as not all channels have the `Kraus` property.